waku/node/v2/waku_types.nim

```diff
@@ -187,15 +191,34 @@ proc generateRequestId*(rng: ref BrHmacDrbgContext): string =
   toHex(bytes)
 
 proc computeIndex*(msg: WakuMessage): Index =
-  ## Takes a WakuMessage and returns its index
+  ## Takes a WakuMessage and returns its Index 
   var ctx: sha256
   ctx.init()
-  if msg.contentTopic != 0: # checks for non-empty contentTopic
-    ctx.update(msg.contentTopic.toBytes()) # converts the topic to bytes
+  ctx.update(msg.contentTopic.toBytes()) # converts the contentTopic to bytes
   ctx.update(msg.payload)
   let digest = ctx.finish() # computes the hash
   ctx.clear()
 
   result.digest = digest
   result.receivedTime = epochTime() # gets the unix timestamp
 
+proc indexComparison* (x, y: Index): int =
```

Comment:
Maybe this belongs on an util folder/file?

Reply:
Do we have such folder or file in the repo (cannot find any)? I can create a util file or move it to the https://github.com/status-im/nim-waku/blob/master/waku/protocol/v2/waku_store.nim as it is the only place the indexComparison is used

Reply:
Moving it to waku_store seems reasonable to me. No util folder yet, can be created if we think it will be used from multiple places.

Reply:
At this point, it is going to be used only for pagination and in the store protocol, cannot think of a situation that it may be required for some other purposes. So, I'll move it to the waku_store